src/Microsoft.TestPlatform.Common/DataCollection/AfterTestRunEndResult.cs

```diff
@@ -24,6 +24,7 @@ private AfterTestRunEndResult()
         AttachmentSets = null!;
         InvokedDataCollectors = null!;
         Metrics = null!;
```

Comment:
Isn't it possible to reuse the metrics dictionary instead of adding an additional argument that we have to propagate everywhere ?